woocommerce/woocommerce-android
# Migrate the scanning library from Google Code Scanner to MLKit

Closes: #9271 


### Description

This PR migrates the scanning library from [Google Code Scanner](https://href.li/?https://developers.google.com/ml-kit/vision/barcode-scanning/code-scanner) to Google's [MLKit](https://href.li/?https://developers.google.com/ml-kit/vision/barcode-scanning).

The reason for the migration can be found here: pdfdoF-3cd-p2

**NOTE:** This PR does not remove the Google Code Scanner library from the codebase yet. The reason is, it's still used in the login flow. We will be migrating to the MLKit in the login flow as well but not in this PR.

### Testing instructions

1. Navigate to the Order listing screen (Orders tab)
2. Click on the barcode icon present in the toolbar
3. Scan the barcode, and make sure it scans the barcode, adds the product
4. Scan any barcode with the SKU not present in your store and ensure the failure cases are handled by displaying a snack bar.
5. Ensure the events are tracked as mentioned in [this PR](https://github.com/woocommerce/woocommerce-android/pull/9110).
6. Navigate to the Order creation screen (Orders tab -> click on "+" button -> click on the barcode icon)
7. Repeat steps 3-4 and ensure the behavior is as expected.


You could also go through the various PRs mentioned in [this one](https://github.com/woocommerce/woocommerce-android/pull/9125) and test the feature as mentioned under the `testing instructions` in each of those PRs.
